dvc/repo/run.py

```diff
@@ -2,7 +2,7 @@
 
 
 def run(
-    self,
+    repo,
```

Comment:
You probably meant to use the same `@revert_ignores_on_failure` decorator for `run()` and `imp()` methods? Or just unresistable cosmetic changes? 🙂 It is worth introducing such either in a separate patch or in a separate PR, so that it doesn't obstruct the view. No need to bother splitting it now, but just worth to keep in mind 🙂

Reply:
You are right, I wanted to introduce annotation here. Reverting